### PR TITLE
Added auto-refresh for SaveList after it is shown

### PR DESCRIPTION
**Pull request by DTanxxx**
Connected the visibility_changed() signal of CanvasItem from SaveManagerGUI node to OnVisibilityChanged() method in SaveManagerGUI script, so that whenever the saves are shown or hidden, SaveList is refreshed.

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
I suppose this takes care of that one specific situation, but I'm looking for a generic approach where the `SaveList` class should self-contain the code for detecting when it has been shown (after not being previously in visible state).

**Comment by DTanxxx** [comment]
Should I instead introduce a new 'prevVisible' property to keep track of the previous visibility state so that I can check in _Process() whether both the property is both 'false' and SaveList IsVisibileInTree() is true, which will trigger refresh operation?

**Comment by hhyyrylainen** [comment]
You should name the variable "wasVisible" but other than that I think that approach is good.

**Comment by DTanxxx** [comment]
Alright! But what did you mean by ‘updating against master’? I have updated my fork against the repository before making commits if that was what you meant.

**Comment by hhyyrylainen** [comment]
Github says this is not up to date against Thrive master branch:
Kuvakaappaus - 2020-08-02 14-16-19

File: project.godot [hunk 1]
```diff
@@ -154,13 +154,13 @@ e_rotate_left={
 }
 e_undo={
 "deadzone": 0.5,
-"events": [ Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":false,"shift":false,"control":true,"meta":false,"command":true,"pressed":false,"scancode":90,"unicode":0,"echo":false,"script":null)
+"events": [ Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":false,"shift":false,"control":true,"meta":true,"command":true,"pressed":false,"scancode":90,"unicode":0,"echo":false,"script":null)
```
Comment (hhyyrylainen) [inline project.godot:157]: I don't think this is a good change.

File: src/saving/NewSaveMenu.cs [hunk 2]
```diff
@@ -35,11 +35,6 @@ public override void _Ready()
         overwriteConfirm = GetNode<ConfirmationDialog>(OverwriteConfirmPath);
     }
 
-    public void RefreshExisting()
-    {
-        saveList.Refresh();
-    }
-
```
Comment (hhyyrylainen) [inline src/saving/NewSaveMenu.cs:42]: This method should be kept just in case a refresh button is added later.

File: src/saving/SaveList.cs [hunk 3]
```diff
@@ -56,6 +58,17 @@ public override void _Process(float delta)
         if (AutoRefreshOnFirstVisible && !refreshedAtLeastOnce && IsVisibleInTree())
         {
             Refresh();
+            wasVisible = true;
+            return;
+        }
+
+        if (!IsVisibleInTree())
```
Comment (hhyyrylainen) [inline src/saving/SaveList.cs:65]: This logic needs to be optional by an exported property, see `AutoRefreshOnFirstVisible` as an example.
Also I feel like this logic should be combined with the first visible refresh check.

**Pull request closed without merging**